internal/scheduler/scheduling_algo_test.go

```diff
@@ -70,24 +70,41 @@ func TestSchedule(t *testing.T) {
 			queuedJobs:               testfixtures.N16Cpu128GiJobs(testfixtures.TestQueue, testfixtures.PriorityClass3, 10),
 			expectedScheduledIndices: []int{0, 1},
 		},
+		"schedule onto executors with some unacknowledged jobs": {
+			schedulingConfig: testfixtures.WithMaxUnacknowledgedJobsPerExecutorConfig(16, testfixtures.TestSchedulingConfig()),
+			executors: []*schedulerobjects.Executor{
+				testfixtures.Test1Node32CoreExecutor("executor1"),
+				testfixtures.Test1Node32CoreExecutor("executor2"),
+			},
+			queues:     []*database.Queue{testfixtures.TestDbQueue()},
+			queuedJobs: testfixtures.N1Cpu4GiJobs(testfixtures.TestQueue, testfixtures.PriorityClass3, 48),
+			scheduledJobsByExecutorIndexAndNodeIndex: map[int]map[int]scheduledJobs{
+				0: {
+					0: scheduledJobs{
+						jobs:         testfixtures.N1Cpu4GiJobs(testfixtures.TestQueue, testfixtures.PriorityClass3, 16),
+						acknowledged: false,
+					},
+				},
+			},
+			expectedScheduledIndices: testfixtures.IntRange(0, 47),
+		},
 		"do not schedule onto executors with too many unacknowledged jobs": {
-			// TODO: This test doesn't look right; we never set MaxUnacknowledgedJobsPerExecutor.
-			schedulingConfig: testfixtures.TestSchedulingConfig(),
+			schedulingConfig: testfixtures.WithMaxUnacknowledgedJobsPerExecutorConfig(15, testfixtures.TestSchedulingConfig()),
```

Comment:
If the executor has 16 unacknowledged jobs and `MaxUnacknowledgedJobsPerExecutor` is set to 15, then we do not give it any more jobs.